heat/core/operations.py

```diff
@@ -385,18 +385,25 @@ def __reduce_op(x, partial_op, reduction_op, **kwargs):
 
     # no further checking needed, sanitize axis will raise the proper exceptions
     axis = stride_tricks.sanitize_axis(x.shape, kwargs.get("axis"))
-    split = x.split
+    if isinstance(axis, int):
+        axis = (axis,)
     keepdim = kwargs.get("keepdim")
+    split = x.split
 
+    # if local tensor is empty, replace it with the identity element
+    if 0 in x.lshape and (axis is None or (x.split in axis)):
+        neutral = kwargs.get("neutral")
+        if neutral is None:
+            neutral = float("nan")
+        neutral_shape = x.lshape[:split] + (1,) + x.lshape[split + 1 :]
+        partial = torch.full(neutral_shape, fill_value=neutral, dtype=x._DNDarray__array.dtype)
+    else:
+        partial = x._DNDarray__array
     if axis is None:
-        partial = partial_op(x._DNDarray__array).reshape(-1)
+        partial = partial_op(partial).reshape(-1)
```

Comment:
Wouldn't it be better to provide the neutral value only in the later call to `Allreduce`? I don't think it's necessary to run the neutral values through the `partial_op`.

Reply:
In principle you're right, however the block of code 402-419 also gives partial (neutral or not) the right shape.

Reply:
I'll think about it a bit longer.

Reply:
@TheSlimvReal I gave it some thought. As I wrote yesterday, in the current implementation the tasks of running partial through partial_op() and assigning partial/neutral the appropriate shape are entangled. I could take them apart, write a nested function for the calculation of the appropriate shapes so that task is independent. It's feels like quite a bit of reshuffling though, and not my highest priority at the moment, so I suggest having a separate issue and dealing with it later. Let me know what you think.

Reply:
@TheSlimvReal @ClaudiaComito i dont think that we need to get rid of it. i think that this is fine. if the node had no data on it to begin with we are not changing the amount of work done as the node is already allocated.